pkg/codec/renamer.go

```diff
@@ -0,0 +1,76 @@
+package codec
+
+import (
+	"reflect"
+	"unicode"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+var pkgPath = reflect.TypeOf(renamer{}).PkgPath()
+
+func NewRenamer(fields map[string]string) Modifier {
+	m := &renamer{
+		modifierBase: modifierBase[string]{
+			fields:           fields,
+			onToOffChainType: map[reflect.Type]reflect.Type{},
+			offToOnChainType: map[reflect.Type]reflect.Type{},
+		},
+	}
+	m.modifyFieldForInput = func(field *reflect.StructField, _, newName string) error {
+		field.Name = newName
+		if unicode.IsLower(rune(field.Name[0])) {
+			field.PkgPath = pkgPath
```

Comment:
I'm not sure I understand the pkgPath stuff.  Does this mean that if we rename a struct in some other package to something new, it will automatically get moved inside the `codec` package?

Reply:
pkgpath is the name of the package that the field is defined in. Since nothing else in this package is defining fields on the modified structs, there won't be another name that's the same.

The intended use will be in [config in this PR](https://github.com/smartcontractkit/chainlink-common/pull/264/files#diff-ede4abc6023e679b5795c220bdd7a8fc49acc42011979f467b5f77b2d1ecf654R87) that drops fields by renaming them to private names.

Reply:
going to merge answers in [ilija42](https://github.com/ilija42)'s thread.